File: plugin/runtime/src/main/java/com/jaynewstrom/json/runtime/internal/FloatJsonAdapter.java
package com.jaynewstrom.json.runtime.internal;

import com.fasterxml.jackson.core.JsonGenerator;
import com.fasterxml.jackson.core.JsonParser;
import com.jaynewstrom.json.runtime.JsonDeserializer;
import com.jaynewstrom.json.runtime.JsonDeserializerFactory;
import com.jaynewstrom.json.runtime.JsonSerializer;
import com.jaynewstrom.json.runtime.JsonSerializerFactory;

import java.io.IOException;

public final class FloatJsonAdapter implements JsonSerializer<Float>, JsonDeserializer<Float> {
    public static final FloatJsonAdapter INSTANCE = new FloatJsonAdapter();

    private FloatJsonAdapter() {
    }

    @Override public void serialize(Float f, JsonGenerator jg, JsonSerializerFactory serializerFactory) throws IOException {
        jg.writeNumber(f);
    }

    @Override public Float deserialize(JsonParser jp, JsonDeserializerFactory deserializerFactory) throws IOException {
        return jp.getFloatValue();
    }

    @Override public Class<?> modelClass() {
        return Float.class;
    }
}

File: plugin/runtime/src/main/java/com/jaynewstrom/jsonDelight/runtime/internal/FloatJsonAdapter.kt
package com.jaynewstrom.jsonDelight.runtime.internal

import com.fasterxml.jackson.core.JsonGenerator
import com.fasterxml.jackson.core.JsonParser
import com.jaynewstrom.jsonDelight.runtime.JsonDeserializer
import com.jaynewstrom.jsonDelight.runtime.JsonDeserializerFactory
import com.jaynewstrom.jsonDelight.runtime.JsonRegistrable
import com.jaynewstrom.jsonDelight.runtime.JsonSerializer
import com.jaynewstrom.jsonDelight.runtime.JsonSerializerFactory

import java.io.IOException

object FloatJsonAdapter : JsonSerializer<Float>, JsonDeserializer<Float>, JsonRegistrable {
    @Throws(IOException::class)
    override fun serialize(value: Float, jg: JsonGenerator, serializerFactory: JsonSerializerFactory) = jg.writeNumber(value)

    @Throws(IOException::class)
    override fun deserialize(jp: JsonParser, deserializerFactory: JsonDeserializerFactory): Float = jp.floatValue

    override fun modelClass(): Class<*> = Float::class.java
}
